Code/Framework/AzCore/AzCore/EBus/EBus.h

```diff
@@ -1999,3 +1999,42 @@ AZ_POP_DISABLE_WARNING
         }
     } // namespace Internal
 }
+
+// The following allow heavily-used busses to be declared extern, in order to speed up compile time where the same header
+// with the same bus is included in many different source files.
+// to use it, declare the EBus extern using DECLARE_EBUS_EXTERN or DECLARE_EBUS_EXTERN_NO_TRAITS in the header file
+// and then use DECLARE_EBUS_INSTANTIATION or DECLARE_EBUS_INSTANTIATION_NO_TRAITS in a file that everything that includes the header
+// will link to (for example, in a static library, dynamic library with export library, or .inl that everyone must include in a compile unit).
+
+// The following must be declared AT GLOBAL SCOPE and the namespace AZ is assumed due to the rule that extern template declarations must occur
+// in their enclosing scope.
+
+//! Externs an EBus class template with both the interface and bus traits arguments
+#define DECLARE_EBUS_EXTERN(a,b) \
+namespace AZ \
+{ \
+   extern template class EBus<a, b>; \
+}
+
+//! Externs an EBus class template using only the interface argument
+//! for both the EBus Interface and BusTraits template parameters
+#define DECLARE_EBUS_EXTERN_NO_TRAITS(a) \
+namespace AZ \
+{ \
+   extern template class EBus<a, a>; \
+}
+
+//! Instantiates an EBus class template with both the interface and bus traits arguments
+#define DECLARE_EBUS_INSTANTIATION(a,b) \
+namespace AZ \
+{ \
+   template class EBus<a, b>; \
+}
+//! Instantiates an EBus class template using only the interface argument
+//! for both the EBus Interface and BusTraits template parameters
+
+#define DECLARE_EBUS_INSTANTIATION_NO_TRAITS(a) \
+namespace AZ \
+{ \
+   template class EBus<a, a>; \
+}
```

Comment:
Suggestion: It seems like these would read a little better in the positive form instead of the negative form.
DECLARE_EBUS_EXTERN(a)
DECLARE_EBUS_AND_TRAITS_EXTERN(a, b)

In reading through the PR, I kept seeing "NO_TRAITS" as possibly something that was getting stripped off for some reason.

Reply:
I can agree with that.  any thoughts from any other reviewers?

Reply:
fwiw, i actually thought about proposing the macro magic route, but backed off from it when i realized the one advantage of the current design is that it makes a mismatch between "DECLARE_EBUS_EXTERN" / "DECLARE_EBUS_INSTANTIATION" a little more obvious, because the names won't line up. a mismatch will be harder to spot if it's just the number of parameters that are different.

Reply:
yeah I think I'll switch it to 
```
DECLARE_EBUS_EXTERN(a)
DECLARE_EBUS_EXTERN_WITH_TRAITS(a,b)
```
I also have to figure out whyt he compile is failing, but not locally

Reply:
I am good with switch to additive terminology.

Reply:
I agree with @mbalfour-amzn suggestion.